### PR TITLE
test: Add test for ORCA

### PR DESCRIPTION
**Pull request by indrajit96**
#### Checklist
- [X] PR title reflects the change and is of format `<commit_type>: <Title>`
- [X] Changes are described in the pull request.
- [X] Related issues are referenced.
- [ ] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [X] Added [test plan](#test-plan) and verified test passes.
- [X] Verified that the PR passes existing CI.
- [X] Verified copyright is correct on all changed files.
- [X] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [ ] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [ ] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [X] test

#### Related PRs:
This is test coverage for https://github.com/triton-inference-server/server/pull/7839

#### Where should the reviewer start?
`test.sh`

#### Test plan:
Load a trt-llm model
Pass orca header in request
Verify orca metrics in response

- CI Pipeline ID:
https://gitlab-master.nvidia.com/dl/dgx/tritonserver/-/pipelines/23834466

### CODE REVIEW

File: qa/L0_orca/orca_http_test.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/python3
+# Copyright 2023-2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (krishung5) [inline qa/L0_orca/orca_http_test.py:2]:
```suggestion
# Copyright 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

File: qa/L0_orca/test.sh [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  echo -e "\n***\n*** ORCA Test Passed\n***"
+else
+    cat $SERVER_LOG
+    cat $CLIENT_LOG
+    echo -e "\n***\n*** ORCA Test FAILED\n***"
+fi
+
+kill_server
+set -e
+exit $RET
```
Comment (krishung5) [inline qa/L0_orca/test.sh:216]: Extra line at the end of file.

File: qa/L0_orca/test.sh [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+RET=0
+BASE_DIR=$(pwd)
+NUM_GPUS=${NUM_GPUS:=1}
```
Comment (krishung5) [inline qa/L0_orca/test.sh:30]: Do we plan to have multi-gpu test variant in the CI as well? Or maybe for future follow up?

File: qa/L0_orca/test.sh [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+if [ "$SERVER_PID" == "0" ]; then
+    echo -e "\n***\n*** Failed to start $SERVER\n***"
+    cat $SERVER_LOG
+    exit 1
+fi
+
+RET=0
+
+python3 $CLIENT_PY
```
Comment (krishung5) [inline qa/L0_orca/test.sh:199]: Can we save the client log? May be helpful for debugging if needed.

**Comment by BenjaminBraunDev** [comment]
Quick thing to note: I just renamed the request header to specify the ORCA format from `endpoint-load-metrics-type` to `endpoint-load-metrics-format`

File: qa/L0_orca/test.sh (hunk 3 above)
Comment (indrajit96) [reply qa/L0_orca/test.sh:30]: multi-gpu won't be needed here as the ORCA are per-request metrics.

File: qa/L0_orca/test.sh [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+ENGINES_DIR="${BASE_DIR}/engines/inflight_batcher_llm/${NUM_GPUS}-gpu"
+TRITON_DIR=${TRITON_DIR:="/opt/tritonserver"}
+SERVER=${TRITON_DIR}/bin/tritonserver
+BACKEND_DIR=${TRITON_DIR}/backends
+SERVER_LOG="${NAME}_server.log"
+SERVER_TIMEOUT=${SERVER_TIMEOUT:=120}
+CLIENT_PY=${BASE_DIR}/orca_http_test.py
+CLIENT_LOG="${NAME}_orca_http_test.log"
+
+function clone_tensorrt_llm_backend_repo {
```
[the remark this answers is not in the document]
Comment (indrajit96) [reply qa/L0_orca/test.sh]: Done

File: qa/L0_orca/test.sh [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    rm -rf $TENSORRTLLM_BACKEND_DIR && mkdir $TENSORRTLLM_BACKEND_DIR
+    apt-get update && apt-get install git-lfs -y --no-install-recommends
+    git clone --single-branch --depth=1 -b ${TENSORRTLLM_BACKEND_REPO_TAG} ${TRITON_REPO_ORG}/tensorrtllm_backend.git $TENSORRTLLM_BACKEND_DIR
+    cd $TENSORRTLLM_BACKEND_DIR && git lfs install && git submodule update --init --recursive
+}
+
+function build_gpt2_base_model {
```
[the remark this answers is not in the document]
Comment (indrajit96) [reply qa/L0_orca/test.sh]: Done

File: qa/L0_orca/test.sh [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+        exit 1
+    fi
+    cd ${GPT_DIR}
+
+    # Convert weights from HF Tranformers to FT format
+    python3 convert_checkpoint.py --model_dir gpt2 --dtype float16 --tp_size ${NUM_GPUS} --output_dir "./c-model/gpt2/${NUM_GPUS}-gpu/"
+    cd ${BASE_DIR}
+}
+
+function build_gpt2_tensorrt_engine {
```
[the remark this answers is not in the document]
Comment (indrajit96) [reply qa/L0_orca/test.sh]: Done

File: qa/L0_orca/test.sh [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+        --remove_input_padding enable \
+        --paged_kv_cache enable \
+        --gemm_plugin float16 \
+        --workers "${NUM_GPUS}" \
+        --output_dir "${ENGINES_DIR}"
+
+    cd ${BASE_DIR}
+}
+
+function replace_config_tags {
```
[the remark this answers is not in the document]
Comment (indrajit96) [reply qa/L0_orca/test.sh]: Done

File: qa/L0_orca/test.sh [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+    replace_config_tags '${engine_dir}' "${ENGINES_DIR}" "${MODEL_REPOSITORY}/tensorrt_llm/config.pbtxt"
+    replace_config_tags '${triton_backend}' "tensorrtllm" "${MODEL_REPOSITORY}/tensorrt_llm/config.pbtxt"
+    replace_config_tags '${max_queue_size}' "0" "${MODEL_REPOSITORY}/tensorrt_llm/config.pbtxt"
+    replace_config_tags '${logits_datatype}' "TYPE_FP32" "${MODEL_REPOSITORY}/tensorrt_llm/config.pbtxt"
+    replace_config_tags '${encoder_input_features_data_type}' "TYPE_FP32" "${MODEL_REPOSITORY}/tensorrt_llm/config.pbtxt"
+}
+
+# Wait until server health endpoint shows ready. Sets WAIT_RET to 0 on
+# success, 1 on failure
+function wait_for_server_ready() {
```
[the remark this answers is not in the document]
Comment (indrajit96) [reply qa/L0_orca/test.sh:87]: The one used here is slightly different and more comprehensive

File: qa/L0_orca/test.sh [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+    if [ "$WAIT_RET" != "0" ]; then
+        # Cleanup
+        kill "${SERVER_PID[@]}" >/dev/null 2>&1 || true
+        echo -e "\n***\n*** Failed to start $SERVER\n***"
+        cat $SERVER_LOG
+        exit 1
+    fi
+}
+
+function kill_server {
```
[the remark this answers is not in the document]
Comment (indrajit96) [reply qa/L0_orca/test.sh:131]: The one used here is slightly different and more comprehensi

**Review by rmccorm4 — APPROVED** [review]

**Review by krishung5 — APPROVED** [review]
